planner/core/task.go

```diff
@@ -1042,8 +1042,23 @@ func (p *PhysicalProjection) GetCost(count float64) float64 {
 }
 
 func (p *PhysicalProjection) attach2Task(tasks ...task) task {
-	// TODO: support projection push down.
-	var t task = tasks[0].convertToRootTask(p.ctx)
+	t := tasks[0].copy()
+	if cop, ok := t.(*copTask); ok {
+		if len(cop.rootTaskConds) == 0 && cop.getStoreType() == kv.TiFlash && expression.CanExprsPushDown(p.ctx.GetSessionVars().StmtCtx, p.Exprs, p.ctx.GetClient(), cop.getStoreType()) {
+			copTask := attachPlan2Task(p, cop)
+			copTask.addCost(p.GetCost(t.count()))
+			return copTask
+		}
+	} else if mpp, ok := t.(*mppTask); ok {
+		if expression.CanExprsPushDown(p.ctx.GetSessionVars().StmtCtx, p.Exprs, p.ctx.GetClient(), kv.TiFlash) {
+			p.SetChildren(mpp.p)
```

Comment:
It's better to implement "attachPlan2Task for mpp task"

Reply:
e... just less then 5 lines code, it does not worth adding a function.

Reply:
It's worthy if we want to involve more operators in the future.

Reply:
then we can updated as needed in the future.